config.go

```diff
@@ -549,6 +551,63 @@ func (s *Settings) GetSender(ctx context.Context, env Environment) (send.Sender,
 	return send.NewConfiguredMultiSender(senders...), nil
 }
 
+// getGithubAppAuth returns app id and app private key if it exists.
+// returns false if it does not have both of those fields.
+func (s *Settings) getGithubAppAuth() (int64, []byte, bool) {
+	key, ok := s.Expansions[GithubAppPrivateKey]
+	if s.AuthConfig.Github != nil && s.AuthConfig.Github.AppId != 0 && ok && key != "" {
+		return s.AuthConfig.Github.AppId, []byte(key), true
+	}
+	return 0, nil, false
+}
+
+func (s *Settings) GetGithubAppClient(caller, owner, repo string, retry utility.HTTPRetryFunction, delay utility.HTTPDelayFunction) *http.Client {
+	appId, key, exists := s.getGithubAppAuth()
+	if !exists {
+		return nil
+	}
+	itr, err := ghinstallation.NewAppsTransport(http.DefaultTransport, appId, key)
```

Comment:
Rather than using the HTTP package's default transporter, can we use the utility client's built-in transporter?
```go
c := utility.GetHTTPClient()
itr, err := ghinstallation.NewAppsTransport(c.Transport, appId, key)
c.Transport = itr
stuff...
```